Debugging.md

```diff
@@ -0,0 +1,83 @@
+
+```
+980d7e80457265230a0bf   "/bin/sh -c ./cppLin…"   About a minute ago   Up About a minute   0.0.0.0:30000->3600/tcp, 0.0.0.0:30001->3601/udp  great_archimedes
+```
+
+In this case, port 30000 on the VM is mapped onto 3600 on the container for TCP whereas 30001 on the VM is mapped on 3601 on the container for UDP. For more information for Docker networking, check [here](https://docs.docker.com/network/).
+
+> In both Windows and Linux, Docker containers are part of the "playfab" Docker network. You can do `docker inspect network playfab` to see information about the network.
+
+### How can I see runtime details for a container?
+
+Once you do `docker ps` you can grab the container name or hash and do `docker inspect <nameOrHash>`. You will see lots of information, like the state of the container, the volume binding on the host VM, port bindings, environment variables passed into the container and more.
+
+### How can I get the logs of my game server?
+
+You can use the command `docker logs <nameOrHash>`. These logs contain everything your app sends to standard output/standard error streams. Bear in mind that these logs exist only for *existing* containers. If your game server crashes, our monitoring process in the VM will delete this container and create a new one.
+
+Consequently, it's a better practice to use GSDK to log from inside your game server. Check [here](https://docs.microsoft.com/en-us/gaming/playfab/features/multiplayer/servers/integrating-game-servers-with-gsdk#logging-with-the-gsdk) for logging with GSDK and [here](https://docs.microsoft.com/en-us/gaming/playfab/features/multiplayer/servers/archiving-and-retrieving-multiplayer-server-logs) for accessing the logs from terminated game servers.
+
+### Can I connect "inside" a running container to see what's going on?
+
+Yup! You can do `docker exec -it <nameOrHash> powershell` on Windows and `docker exec -it <nameOrHash> bash` on Linux. There, you get access to a command line process in the container where you can issue native commands to debug/diagnose issues.
+
+> For this command to work on Linux, Bash shell must be installed in your base container. If it isn't, you can use Bourne shell by running `docker exec -it <nameOrHash> sh`.
+
+### How can I see the ports that are open on my container?
+
+Once you are connected "inside" the container, you can use `netstat -ano` on Windows and `netstat -tulpn` on Linux when you have a command line process inside your container (check the previous instruction).
+
+> If this command doesn't work on Linux, try installing netstat by `apt update && apt install net-tools`
+> On Linux, you can use `nestatst -tulpn` inside the VM to see that the ports are indeed open. Recall that VM ports will be different than container ports, you can see this port mapping with `docker ps` for your containers.
+
+## Linux specific instructions
+
+### On Linux, how can I see diagnostics about my container?
```

Comment:
>On Linux [](start = 4, length = 8)

On Linux can probably be removed from the questions since they are redundant (the section says Linux specific instructions)